src/phpDocumentor/Uri.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor;
+
+/**
+ * Class Uri
+ *
+ * @package phpDocumentor
+ */
+final class Uri
+{
+    /**
+     * @var string
+     */
+    private $uri;
+
+    /**
+     * @param string $uri
+     */
+    public function __construct($uri)
+    {
+        $this->validateString($uri);
+
+        $uri = $this->checkScheme($uri);
+
+        $this->validateUri($uri);
+
+        $this->uri = $uri;
+    }
+
+    /**
+     * @return string
+     */
+    public function __toString()
+    {
+        return $this->uri;
+    }
+
+    /**
+     * Checks if $uri is of type string.
+     *
+     * @param  mixed $uri
+     * @throws \InvalidArgumentException
```

Comment:
Can you add a description with the exception that describes when this is thrown?